src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,209 @@
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getRequest = "SELECT cars.id, model, manufacturer_id, "
+                          + "manufacturers.id, name, country "
+                          + "FROM cars INNER JOIN manufacturers "
+                          + "ON manufacturers.id = cars.manufacturer_id "
+                          + "WHERE cars.id = ? AND cars.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection
+                        .prepareStatement(getRequest, Statement.RETURN_GENERATED_KEYS)) {
+            getCarStatement.setObject(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car id: " + id + " from DB", e);
+        }
+        if (car != null) {
+            car.setDrivers(getCarDrivers(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCarsRequest = "SELECT cars.id, model, manufacturer_id, "
+                + "manufacturers.id, name, country "
+                + "FROM cars INNER JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection
+                        .prepareStatement(getAllCarsRequest)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get cars from DB", e);
+        }
+        cars.forEach(c -> c.setDrivers(getCarDrivers(c.getId())));
+        return cars;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarByDriverRequest = "SELECT car_id FROM car_drivers WHERE driver_id = ?";
+        List<Car> cars = new ArrayList<>();
+        List<Long> carsId = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsByDriverStatement = connection
+                        .prepareStatement(getCarByDriverRequest, Statement.RETURN_GENERATED_KEYS)) {
+            getCarsByDriverStatement.setObject(1, driverId);
+            ResultSet resultSet = getCarsByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                Long carId = resultSet.getObject("car_id", Long.class);
+                carsId.add(carId);
```

Comment:
we can use the same logic as in getAll method. Let's init cars without drivers here, and after try catch in forEach init drivers for each car

Reply:
Why it will be better? In any way here will be two different select request to get car fields and driver id.